File: experiments/runs/2026-09-20_07-22-29_claim-office-java-example-mapping_baseline-inline-tdd-v1-cc_opus-5-no-thinking/src/main/java/PriceList.java
import java.util.Map;

/** The MHPCO price list: insurance values and base premiums per item type. */
final class PriceList {

    private static final Map<String, Entry> ENTRIES = Map.of(
            "sword", new Entry(1000, 100),
            "amulet", new Entry(600, 60),
            "staff", new Entry(800, 80),
            "potion", new Entry(400, 40),
            "rune", new Entry(250, 25),
            "moonstone", new Entry(250, 25));

    private PriceList() {
    }

    static int basePremium(String type) {
        return entry(type).basePremium();
    }

    static int insuranceValue(String type) {
        return entry(type).insuranceValue();
    }

    private static Entry entry(String type) {
        Entry entry = ENTRIES.get(type);
        if (entry == null) {
            throw new ClaimOfficeException("unknown item type: " + type);
        }
        return entry;
    }

    private record Entry(int insuranceValue, int basePremium) {
    }
}

File: experiments/runs/2026-09-23_17-35-42_claim-office-example-mapping_exact-ptdd-v1.1-refactor-subagent-cc_opus-5-5-no-thinking/src/priceList.ts
/** MHPCO price list: insurance value and base premium of every insurable item type. */

export interface ItemPrice {
  insuranceValue: number;
  basePremium: number;
}

const MAIN_ITEM_PRICES: Record<string, ItemPrice> = {
  sword: { insuranceValue: 1000, basePremium: 100 },
  amulet: { insuranceValue: 600, basePremium: 60 },
  staff: { insuranceValue: 800, basePremium: 80 },
  potion: { insuranceValue: 400, basePremium: 40 },
};

/** Components are priced and insured per piece; a block of alike components has a special base premium. */
const COMPONENT_PRICE: ItemPrice = { insuranceValue: 250, basePremium: 25 };
export const COMPONENT_BLOCK_SIZE = 3;
export const COMPONENT_BLOCK_BASE_PREMIUM = 60;
const COMPONENT_TYPES = new Set(["rune", "moonstone"]);

/** Components (runes, moonstones, ...) are priced and insured per piece rather than as main items. */
export function isComponent(type: string): boolean {
  return COMPONENT_TYPES.has(type);
}

/**
 * The price-list entry of an item type: components share one entry, main items have their own.
 * Item types missing from the price list are not insurable and are rejected.
 */
export function priceOf(type: string): ItemPrice {
  if (isComponent(type)) return COMPONENT_PRICE;
  const price = MAIN_ITEM_PRICES[type] as ItemPrice | undefined;
  if (price === undefined) throw new Error(`Unknown item type '${type}': not on the MHPCO price list`);
  return price;
}
